src/layers.jl

```diff
@@ -1,5 +1,7 @@
 module Layers
-export Layer, next, top_layer, insert
+export Layer, next, top_layer, insert, insert_default!, remove_default!
+
+const EXTRA_LAYERS = Set{Tuple{Union{UnionAll, Type{Union{}}}, UnionAll}}()
```

Comment:
Hmmmm.....I don't quite understand the eltype of the `Set` here; why all the Tuple/Union/UnionAll dance?

Reply:
I was trying to keep all the type parameters as concrete as possible. The `Set` elements are `Tuple`s with two elements: the first is the before layer, which can be either a `Type{<:Layer}` or `Type{Union{}}` (in the case that we insert a layer as the last layer), and the second is the custom layer, which is always a `Type{<:Layer}`. Since layer types are always parametric, a non-parameterized layer type is a `UnionAll`. Does that make sense? Maybe it's totally unnecessary to parametrize the `Set` like this.